src/Data/Nat/DivMod.agda

```diff
@@ -255,6 +299,88 @@ m*n/m*o≡n/o m@(suc m-1) n o = helper (<-wellFounded n)
     n / o                         ∎
     where n∸o<n = ∸-monoʳ-< (n≢0⇒n>0 (≢-nonZero⁻¹ o)) (≮⇒≥ n≮o)
 
+m*n/o*n≡m/o : ∀ m n o .⦃ _ : NonZero o ⦄ ⦃ _ : NonZero (o * n) ⦄ →
+              m * n / (o * n) ≡ m / o
+m*n/o*n≡m/o m n o ⦃ _ ⦄ ⦃ o*n≢0 ⦄ = begin-equality
+  m * n / (o * n) ≡⟨ /-congˡ (*-comm m n) ⟩
+  n * m / (o * n) ≡⟨ /-congʳ (*-comm o n) ⟩
+  n * m / (n * o) ≡⟨ m*n/m*o≡n/o n m o ⟩
+  m / o           ∎
+  where instance n*o≢0 = subst NonZero (*-comm o n) o*n≢0
+
+m<n*o⇒m/o<n : ∀ {m n o} .⦃ _ : NonZero o ⦄ → m < n * o → m / o < n
+m<n*o⇒m/o<n {m} {suc n} {o} m<n*o with m <? o
+... | yes m<o = begin-strict
+  m / o ≡⟨ m<n⇒m/n≡0 m<o ⟩
+  0     <⟨ s≤s z≤n ⟩
+  suc n ∎
+... | no m≮o = begin-strict
+  m / o             ≡⟨ m/n≡1+[m∸n]/n (≮⇒≥ m≮o) ⟩
+  suc ((m ∸ o) / o) <⟨ s≤s (m<n*o⇒m/o<n lem) ⟩
+  suc n             ∎
+  where
+  lem : m ∸ o < n * o
+  lem = begin-strict
+    m ∸ o         <⟨ ∸-monoˡ-< m<n*o (≮⇒≥ m≮o) ⟩
+    o + n * o ∸ o ≡⟨ m+n∸m≡n o (n * o) ⟩
+    n * o         ∎
+
+[m∸n]/n≡m/n∸1 : ∀ m n .⦃ _ : NonZero n ⦄ → (m ∸ n) / n ≡ pred (m / n)
+[m∸n]/n≡m/n∸1 m n with m <? n
+... | yes m<n = begin-equality
+  (m ∸ n) / n  ≡⟨ m<n⇒m/n≡0 (<-transʳ (m∸n≤m m n) m<n) ⟩
+  0            ≡⟨⟩
+  0 ∸ 1        ≡˘⟨ cong (_∸ 1) (m<n⇒m/n≡0 m<n) ⟩
+  m / n ∸ 1    ≡⟨⟩
+  pred (m / n) ∎
+... | no m≮n = begin-equality
+  (m ∸ n) / n           ≡⟨⟩
+  suc ((m ∸ n) / n) ∸ 1 ≡˘⟨ cong (_∸ 1) (m/n≡1+[m∸n]/n (≮⇒≥ m≮n)) ⟩
+  m / n ∸ 1             ≡⟨⟩
+  pred (m / n)          ∎
+
+[m∸n*o]/o≡m/o∸n : ∀ m n o .⦃ _ : NonZero o ⦄ → (m ∸ n * o) / o ≡ m / o ∸ n
+[m∸n*o]/o≡m/o∸n m zero    o = refl
+[m∸n*o]/o≡m/o∸n m (suc n) o = begin-equality
+  (m ∸ (o + n * o)) / o ≡˘⟨ /-congˡ (∸-+-assoc m o (n * o)) ⟩
+  (m ∸ o ∸ n * o) / o   ≡⟨ [m∸n*o]/o≡m/o∸n (m ∸ o) n o ⟩
+  (m ∸ o) / o ∸ n       ≡⟨ cong (_∸ n) ([m∸n]/n≡m/n∸1 m o) ⟩
+  m / o ∸ 1 ∸ n         ≡⟨ ∸-+-assoc (m / o) 1 n ⟩
+  m / o ∸ suc n         ∎
+
+m/n/o≡m/[n*o] : ∀ m n o .⦃ _ : NonZero n ⦄ .⦃ _ : NonZero o ⦄
+                .⦃ _ : NonZero (n * o) ⦄ → m / n / o ≡ m / (n * o)
+m/n/o≡m/[n*o] m n o = begin-equality
+  m / n / o                             ≡⟨ /-congˡ {o = o} (/-congˡ (m≡m%n+[m/n]*n m n*o)) ⟩
+  (m % n*o + m / n*o * n*o) / n / o     ≡⟨ /-congˡ (+-distrib-/-∣ʳ (m % n*o) lem₁) ⟩
+  (m % n*o / n + m / n*o * n*o / n) / o ≡⟨ cong (λ # → (m % n*o / n + #) / o) lem₂ ⟩
```

Comment:
Just to say I really like the use of `#` :smile:

Reply:
Thanks! It makes it stick out, doesn't it? Much easier to see where the substitution happens, I think.